fili-core/src/test/groovy/com/yahoo/bard/webservice/web/SlicesApiRequestSpec.groovy

```diff
@@ -96,13 +107,17 @@ class SlicesApiRequestSpec extends Specification {
                 metricsResult.add(row)
             }
         }
+        Set<SortedMap<DateTime, Map<String, SegmentInfo>>> sliceMetadata = dataSourceMetadataService.getTableSegments(
+                Collections.singleton(table)
+        );
 
         Map<String, Object> expected = new LinkedHashMap<>()
         expected.put("name", name);
         expected.put("timeGrain", fullDictionary.get(name).getTimeGrain().getName());
         expected.put("timeZone", "UTC");
         expected.put("dimensions", dimensionsResult);
         expected.put("metrics", metricsResult);
+        expected.put("segmentInfo", SlicesApiRequest.generateSegmentMetadataView(sliceMetadata))
```

Comment:
We should add a comment though indicating that this test is not intended as a test of `generateSegmentMetadataView` but only that the results get added to the payload.